src/Feature/Account/code/Controllers/AccountsController.cs

```diff
@@ -0,0 +1,146 @@
+
+        private readonly ITrackingService trackingService;
+
+        private readonly IVisitorContext visitorContext;
+
+        public AccountsController(
+            IAccountService accountService,
+            IAccountEntityMapper entityMapper,
+            IVisitorContext visitorContext,
+            ITrackingService trackingService)
+        {
+            Assert.ArgumentNotNull(accountService, nameof(accountService));
+            Assert.ArgumentNotNull(entityMapper, nameof(entityMapper));
+            Assert.ArgumentNotNull(visitorContext, nameof(visitorContext));
+            Assert.ArgumentNotNull(trackingService, nameof(trackingService));
+
+            this.accountService = accountService;
+            this.entityMapper = entityMapper;
+            this.visitorContext = visitorContext;
+            this.trackingService = trackingService;
+        }
+
+        [HttpPost]
+        [Authorize]
+        [ActionName("address")]
+        public ActionResult AddAddress(AddressRequest request)
+        {
+            return this.Execute(
+                () => this.accountService.AddAddress(
+                    this.visitorContext.ContactId,
+                    this.entityMapper.Map<AddressRequest, Address>(request)));
+        }
+
+        [HttpPut]
+        [Authorize]
+        [ActionName("password")]
+        public ActionResult ChangePassword(ChangePasswordRequest request)
+        {
+            return this.Execute(
+                () =>
+                {
+                    this.trackingService.EnsureTracker();
+                    return this.accountService.ChangePassword(request.Email, request.NewPassword, request.OldPassword);
+                });
+        }
+
+        [HttpPost]
+        [AllowAnonymous]
+        [ActionName("account")]
+        public ActionResult CreateAccount(CreateAccountRequest requests)
+        {
+            return this.Execute(
+                () =>
+                {
+                    this.trackingService.EnsureTracker();
+                    return this.accountService.CreateAccount(
+                        requests.Email,
+                        requests.FirstName,
+                        requests.LastName,
+                        requests.Password);
+                });
+        }
+
+        [HttpGet]
+        [AllowAnonymous]
+        [ActionName("address")]
+        public ActionResult GetAddress()
+        {
+            return this.Execute(() => this.accountService.GetAddress(this.visitorContext.ContactId));
+        }
+
+        [HttpDelete]
+        [Authorize]
+        [ActionName("address")]
+        public ActionResult RemoveAddress(AddressRequest request)
+        {
+            return this.Execute(
+                () => this.accountService.RemoveAddress(
+                    this.visitorContext.ContactId,
+                    this.entityMapper.Map<AddressRequest, Address>(request)));
+        }
+
+        [HttpPut]
+        [Authorize]
```

Comment:
Does it redirect to login page or outputs json with error 401/403 code?

Reply:
Out put with error 404